app/scripts/metamask-controller.js

```diff
@@ -4552,4 +4557,30 @@ export default class MetamaskController extends EventEmitter {
       }
     }
   };
+
+  async securityProviderRequest(requestData, methodName) {
+    const { currentLocale, transactionSecurityCheckEnabled } =
+      this.preferencesController.store.getState();
+
+    const chainId = Number(
+      hexToDecimal(this.networkController.getCurrentChainId()),
+    );
+
+    if (transactionSecurityCheckEnabled) {
+      try {
+        const { flagAsDangerous } = await securityProviderCheck(
+          requestData,
+          methodName,
+          chainId,
+          currentLocale,
+        );
+        return await flagAsDangerous;
```

Comment:
Not sure why is this await `await flagAsDangerous` required.

Reply:
Fixed here: 352f1d9054aefd84dd03b992873db4bf61cebc39